lib/compiler-singlepass/src/machine.rs

```diff
@@ -20,11 +20,22 @@ pub(crate) struct Machine {
     locals_offset: MachineStackOffset,
 }
 
+/// Returns an u32 that has as 1 bits the ones matching registers passed as parameters
+macro_rules! bitset_of_regs {
+    ($( $r:expr ),*) => {{
+        $( (1u32 << ($r as u32)) )|*
+    }}
+}
+
 impl Machine {
     pub(crate) fn new() -> Self {
+        // Note: the below asserts are because we currently use u16 for used_gprs and used_xmms
+        // Feel free to increase the number in this assert by making them bigger if needed
+        assert!(GPR::num_gprs() <= 32);
```

Comment:
Assert in a const context for static asserts. 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ca47ea64fb85d4205a79b31d1121d312